panic occuerred when the specified datadir does not exists

```
$ got install --datadir tmp tennashi/got
panic: open tmp/package.lock.json: no such file or directory

goroutine 1 [running]:
github.com/tennashi/got.NewJSONStore(0xc000074870, 0xc000139b30, 0xc000139b48, 0x4f515f, 0xc00001a330)
	/home/tennashi/src/github.com/tennashi/got/json_store.go:28 +0x1b7
github.com/tennashi/got.NewInstalledPackageRepository(0xc000074870, 0xc000139bb0, 0x2, 0xc00001a330, 0x15)
	/home/tennashi/src/github.com/tennashi/got/repository.go:41 +0x65
github.com/tennashi/got.NewInstallCommand(0xc000074870, 0xc000139c98, 0x5, 0xc00001a000, 0x12)
	/home/tennashi/src/github.com/tennashi/got/install.go:52 +0x1d7
github.com/tennashi/got/cmd/got/internal/cli.NewInstallCommand.func2(0xc000024780, 0x0, 0x0)
	/home/tennashi/src/github.com/tennashi/got/cmd/got/internal/cli/install.go:79 +0x1fd
github.com/urfave/cli/v2.(*Command).Run(0xc00016a000, 0xc000024540, 0x0, 0x0)
	/home/tennashi/.asdf/installs/golang/1.16.5/packages/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x4dd
github.com/urfave/cli/v2.(*App).RunContext(0xc00007cea0, 0x6922d8, 0xc00001e0b8, 0xc0000100a0, 0x5, 0x5, 0x0, 0x0)
	/home/tennashi/.asdf/installs/golang/1.16.5/packages/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x810
github.com/urfave/cli/v2.(*App).Run(...)
	/home/tennashi/.asdf/installs/golang/1.16.5/packages/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
github.com/tennashi/got/cmd/got/internal/cli.Run(0xc00006a058)
	/home/tennashi/src/github.com/tennashi/got/cmd/got/internal/cli/cli.go:35 +0x337
main.main()
	/home/tennashi/src/github.com/tennashi/got/cmd/got/main.go:10 +0x25
```

--- a/json_store.go
+++ b/json_store.go
@@ -3,6 +3,8 @@ package got
 import (
 	"encoding/json"
 	"log"
+	"os"
+	"path/filepath"
 
 	"github.com/schollz/jsonstore"
 )
@@ -21,6 +23,10 @@ type JSONStore struct {
 func NewJSONStore(ioStream *IOStream, cfg *JSONStoreConfig) (*JSONStore, error) {
 	if cfg.FilePath == "" {
 		return nil, &InvalidParamError{Message: "file path must not be empty"}
+	}
+
+	if err := os.MkdirAll(filepath.Dir(cfg.FilePath), 0755); err != nil {
+		return nil, err
 	}
 
 	if _, err := jsonstore.Open(cfg.FilePath); err != nil {

